src/protocol.h

```diff
@@ -361,43 +362,97 @@ class CAddress : public CService
 {
     static constexpr uint32_t TIME_INIT{100000000};
 
+    /** Historically, CAddress disk serialization stored the CLIENT_VERSION, optionally OR'ed with
+     *  the ADDRV2_FORMAT flag to indicate V2 serialization. The first field has since been
+     *  disentangled from client versioning, and now instead:
+     *  - The low bits (masked by DISK_VERSION_IGNORE_MASK) store the fixed value DISK_VERSION_INIT,
+     *    (in case any code exists that treats it as a client version) but are ignored on
+     *    deserialization.
+     *  - The high bits (masked by ~DISK_VERSION_IGNORE_MASK) store actual serialization information.
+     *    Only 0 or DISK_VERSION_ADDRV2 (equal to the historical ADDRV2_FORMAT) are valid now, and
+     *    any other value triggers a deserialization failure. Other values can be added later if
+     *    needed.
+     *
+     *  For disk deserialization, ADDRV2_FORMAT signals that ADDRV2 deserialization is permitted,
+     *  but the actual format is determined by the high bits in the stored version field.
+     *  For network serialization ADDRV2_FORMAT determines the actual format used (as it has no
+     *  embedded version number).
+     */
```

Comment:
Maybe also mention what happens during disk serialization and network deserialization:

For disk serialization, ADDRV2_FORMAT in the stream version indicates a write in that format
For network deserialization ADDRV2_FORMAT in the stream version indicates that the data is expected to be in that format

Reply:
I feel that's pretty much already mentioned (just above, and in other places). I've tried to improve the comments further.